src/herder/SurgePricingUtils.cpp

```diff
@@ -160,6 +160,41 @@ SurgePricingPriorityQueue::getMostTopTxsWithinLimits(
     SurgePricingPriorityQueue queue(
         /* isHighestPriority */ true, laneConfig,
         stellar::rand_uniform<size_t>(0, std::numeric_limits<size_t>::max()));
+
+    bool allFit = true;
```

Comment:
This optimization looks ok implementation-wise, but I wonder what is the actual impact. Do we have many ledgers where classic phase is not surge priced?
If surge pricing is an issue, maybe we should also look into optimizing it?

Reply:
Marking as resolved as per offline conversation - this seems to be useful for the loadgen testing and doesn't look too cumbersome to maintain.